core/application/impl/configuration_storage_impl.hpp

```diff
@@ -19,22 +19,72 @@ namespace kagome::application {
 
     ~ConfigurationStorageImpl() override = default;
 
-    GenesisRawConfig getGenesis() const override;
+    const std::string &name() const override {
+      return name_;
+    }
+
+    const std::string &id() const override {
+      return id_;
+    }
+
+    const std::string &chainType() const override {
+      return chain_type_;
+    }
+
     network::PeerList getBootNodes() const override;
 
+    const std::vector<std::pair<std::string, size_t>> &telemetryEndpoints()
+        const override {
+      return telemetry_endpoints_;
+    }
+
+    const std::string &protocolId() const override {
+      return protocol_id_;
+    }
+
+    boost::optional<std::reference_wrapper<const std::string>> getProperty(
+        const std::string &property) const override {
+      auto it = properties_.find(property);
+      if (it != properties_.end()) {
+        return {{it->second}};
+      }
+      return boost::none;
+    }
+
+    const std::set<primitives::BlockHash> &forkBlocks() const override {
+      return fork_blocks_;
+    }
+
+    const std::set<primitives::BlockHash> &badBlocks() const override {
+      return bad_blocks_;
+    }
+
+    boost::optional<std::string> consensusEngine() const override {
+      return consensus_engine_;
+    }
+
+    GenesisRawConfig getGenesis() const override;
```

Comment:
Maybe put impl here, since the other methods-getters implemented in header?: )

Reply:
Why not? Got it